app/features/build_runner/build_runner.rb

```diff
@@ -91,6 +91,27 @@ def setup
       not_implemented(__method__)
     end
 
+    def fail_build!(start_time:)
+      duration = Time.now - start_time
+      current_build.duration = duration
+      current_build.status = :failure # TODO: also handle failure
+      save_build_status!
+    end
+
+    def run_completed_ensure
+      # Make sure to notify the listeners that the build is over
+      new_row(
```

Comment:
@KrauseFx is calling `new_row` ok if called from `fail_build!` instead of from the build completion flow? `fastlane_build_runner` won't have connected any outputs AFAICT

Reply:
I think it should be fine. It is important that `new_row` is called exactly once with `:last_message` no matter if the build suceeded or failed